xunit-autolabeler-v2/ast_parser/core/polyglot_parser.py

```diff
@@ -35,6 +35,7 @@ def add_children_drift_data(
     Note:
         This method should be called once all other parsing is complete
     """
+    recursed_nodes = set()
 
     def _recursor(method: pdd.PolyglotDriftData) -> pdd.PolyglotDriftData:
```

Comment:
It doesn't really return the PolyglotDriftData class, but instead modifies the input, right? It's weird the type checker doesn't catch this.

Reply:
Good catch - changed return type to `None`.